lib/components/popups/course-search/query-components/Form.tsx

```diff
@@ -168,179 +135,18 @@ const Form: FC<{ setSearching: (searching: boolean) => void }> = (props) => {
       levels: searchFilters.levels,
     };
 
-    props.setSearching(true);
+    setSearching(true);
+    dispatch(updateRetrievedCourses([]));
 
     if (searchTerm.length > 0) {
       // Search with half second debounce.
-      const search = setTimeout(
-        performSmartSearch(searchTerm, extras, searchTerm.length),
-        500,
-      );
+      const search = setTimeout(performSmartSearch(extras), 500);
       return () => clearTimeout(search);
     } else {
-      find(extras).then((found) => dispatch(updateRetrievedCourses(found[0])));
+      setSearching(false);
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [searchTerm, searchFilters]);
-
-  /**
```

Comment:
Functions below are no longer necessary. All filtering is done in backend as a db query.